src/webots/gui/WbMainWindow.cpp

```diff
@@ -1956,8 +1956,11 @@ void WbMainWindow::newWorld() {
   WbNewWorldWizard wizard(this);
   wizard.exec();
   simulationState->resumeSimulation();
-  if (!wizard.fileName().isEmpty())
-    loadWorld(WbProject::current()->worldsPath() + wizard.fileName());
+  if (!wizard.fileName().isEmpty()) {
+    QFile newWorldFile(wizard.fileName());
+    if (newWorldFile.exists())
+      loadWorld(WbProject::current()->worldsPath() + wizard.fileName());
+  }
```

Comment:
There is no need to define a variable here that is used only once, additionally QFile` class has a static method to check the existence of a file without having to create an instance of `QFile`.
Moreover the nested IF blocks should be merged:
```
if (!wizard.fileName().isEmpty() && QFile::exists(wizard.fileName())
    loadWorld(WbProject::current()->worldsPath() + wizard.fileName());
```